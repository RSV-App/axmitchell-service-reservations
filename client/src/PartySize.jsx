import React from 'react';
import PropTypes from 'prop-types';

import '../../public/styles.css';

class PartySize extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      size: 2,
    };

    this.handleSizeChange = this.handleSizeChange.bind(this);
    this.onChangeFuncs = this.onChangeFuncs.bind(this);
  }

  handleSizeChange(e) {
    this.props.handleSize(e);
    this.setState({
      size: e.target.value,
    });
  }

  onChangeFuncs(e) {
    this.handleSizeChange(e);
    this.props.getBtnBack();
  }

  render() {
    const size = [];
    for (let i = 1; i <= 20; i++) {
      size.push(i);
    }

    const partySize = size.map(party => <option className="party-size-selected" key={party} value={party}>{party}</option>);

    return (
      <div className="party-size-wrapper">
        <div className="party-size-title">Party Size</div>
        <select className="party-size-list" defaultValue="2" onChange={(e) => { this.onChangeFuncs(e); }}>
          {partySize}
        </select>
      </div>
    );
  }
}

PartySize.propTypes = {
  handleSize: PropTypes.func,
  getBtnBack: PropTypes.func,
};

export default PartySize;
