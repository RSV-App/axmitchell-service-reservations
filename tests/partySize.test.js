import React from 'react';
import {
  shallow, configure, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import PartySize from '../client/src/PartySize.jsx';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer
    .create(<PartySize />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Party Size Rendering', () => {
  const wrapper = shallow(<PartySize />);
  it('renders a select drop down list', () => {
    const list = wrapper.find('.party-size-list');
    expect(list.length).toBe(1);
    expect(list.find('.party-size-selected').length).toBe(20);
  });

  it('should have default party size be 2', () => {
    expect(wrapper.state('size')).toBe(2);
  });
});
