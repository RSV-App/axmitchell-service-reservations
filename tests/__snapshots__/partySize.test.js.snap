// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="party-size-wrapper"
>
  <div
    className="party-size-title"
  >
    Party Size
  </div>
  <select
    className="party-size-list"
    defaultValue="2"
    onChange={[Function]}
  >
    <option
      className="party-size-selected"
      value={1}
    >
      1
    </option>
    <option
      className="party-size-selected"
      value={2}
    >
      2
    </option>
    <option
      className="party-size-selected"
      value={3}
    >
      3
    </option>
    <option
      className="party-size-selected"
      value={4}
    >
      4
    </option>
    <option
      className="party-size-selected"
      value={5}
    >
      5
    </option>
    <option
      className="party-size-selected"
      value={6}
    >
      6
    </option>
    <option
      className="party-size-selected"
      value={7}
    >
      7
    </option>
    <option
      className="party-size-selected"
      value={8}
    >
      8
    </option>
    <option
      className="party-size-selected"
      value={9}
    >
      9
    </option>
    <option
      className="party-size-selected"
      value={10}
    >
      10
    </option>
    <option
      className="party-size-selected"
      value={11}
    >
      11
    </option>
    <option
      className="party-size-selected"
      value={12}
    >
      12
    </option>
    <option
      className="party-size-selected"
      value={13}
    >
      13
    </option>
    <option
      className="party-size-selected"
      value={14}
    >
      14
    </option>
    <option
      className="party-size-selected"
      value={15}
    >
      15
    </option>
    <option
      className="party-size-selected"
      value={16}
    >
      16
    </option>
    <option
      className="party-size-selected"
      value={17}
    >
      17
    </option>
    <option
      className="party-size-selected"
      value={18}
    >
      18
    </option>
    <option
      className="party-size-selected"
      value={19}
    >
      19
    </option>
    <option
      className="party-size-selected"
      value={20}
    >
      20
    </option>
  </select>
</div>
`;
